src/font/fdue.rs

```diff
@@ -51,6 +77,32 @@ impl Font {
 
         Ok(Font { inner })
     }
+
+    fn try_find_best_font(name: &str) -> Vec<String> {
+        const COUNT_MATCHES: usize = 5;
+
+        FONTCONFIG
+            .cache
+            .list()
+            .keys()
+            .filter(|v| v.name.is_some())
+            .map(|font| {
+                let current = font.name.as_ref().unwrap().as_str();
+                if let Some(matching) = best_match(name, current) {
+                    return Some(FuzzyResult {
+                        text: current,
+                        match_: matching,
+                    });
+                }
+                None
+            })
+            .flatten()
+            .collect::<BinaryHeap<FuzzyResult>>()
+            .into_iter()
+            .take(COUNT_MATCHES)
```

Comment:
Apparently that's not you're looking for: `BinaryHeap::IntoIter` isn't working as consecutive `BinaryHeap::pop` calls. [Playground example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=27e6ef1d4d4b530ae9e133e979c5f280).

Reply:
Thank you. I didn't know that. It is very counter-intuitive for me.